core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -399,35 +400,17 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                         "Cannot parse suggestion response without type information. Set [" + RestSearchAction.TYPED_KEYS_PARAM
                                 + "] parameter on the request to ensure the type information is added to the response output");
             }
-            Suggestion suggestion = null;
-            Function<XContentParser, Entry> entryParser = null;
-            // the "size" parameter and the SortBy for TermSuggestion cannot be parsed from the response, use default values
-            // TODO investigate if we can use NamedXContentRegistry instead of this switch
-            switch (type) {
-                case Suggestion.NAME:
```

Comment:
With my limited knowledge of suggesters, I tend to think that we can remove the "generic" parsing case: we're not sure if custom suggesters really need this and even if it's the case a custom parser will have to be registered in the named xcontent registry anyway.